x/concentrated-liquidity/pool.go

```diff
@@ -23,7 +23,12 @@ func NewConcentratedLiquidityPool(poolId uint64) Pool {
 		Id:      poolId,
 	}
 
-	return *pool
+	err := pool.SetInitialPoolAssets(denoms)
+	if err != nil {
+		return Pool{}, err
+	}
+
```

Comment:
I think we should make it so that we store the pool itself.

(Not in the scope of this PR but in subsequent PRs, @mattverse would create PR for this)